src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -489,65 +489,68 @@ Field buildField(ColumnMetadata column) {
         break;
-
-        // TODO: not currently supported in convertRowToArrow
-        //      case TIMESTAMP_TZ:
-        //        switch (physicalType) {
-        //          case SB8:
-        //            arrowType = Types.MinorType.STRUCT.getType();
-        //            FieldType fieldTypeEpoch =
-        //                new FieldType(true, Types.MinorType.BIGINT.getType(), null, metadata);
-        //            FieldType fieldTypeTimezone =
-        //                new FieldType(true, Types.MinorType.INT.getType(), null, metadata);
-        //            Field fieldEpoch = new Field(Constants.FIELD_NAME_EPOCH, fieldTypeEpoch,
-        // null);
-        //            Field fieldTimezone = new Field(Constants.FIELD_NAME_FRACTION,
-        // fieldTypeTimezone, null);
-        //            children = new LinkedList<>();
-        //            children.add(fieldEpoch);
-        //            children.add(fieldTimezone);
-        //            break;
-        //          case SB16:
-        //            arrowType = Types.MinorType.STRUCT.getType();
-        //            fieldTypeEpoch = new FieldType(true, Types.MinorType.BIGINT.getType(), null,
-        // metadata);
-        //            FieldType fieldTypeFraction =
-        //                new FieldType(true, Types.MinorType.INT.getType(), null, metadata);
-        //            fieldTypeTimezone = new FieldType(true, Types.MinorType.INT.getType(), null,
-        // metadata);
-        //            fieldEpoch = new Field(Constants.FIELD_NAME_EPOCH, fieldTypeEpoch, null);
-        //            Field fieldFraction = new Field(Constants.FIELD_NAME_FRACTION,
-        // fieldTypeFraction, null);
-        //            fieldTimezone =
-        //                new Field(Constants.FIELD_NAME_TIME_ZONE_INDEX, fieldTypeTimezone, null);
-        //
-        //            children = new LinkedList<>();
-        //            children.add(fieldEpoch);
-        //            children.add(fieldFraction);
-        //            children.add(fieldTimezone);
-        //            break;
-        //          default:
-        //            throw new SFException(
-        //                ErrorCode.UNKNOWN_DATA_TYPE,
-        //                "Unknown physical type for TIMESTAMP_TZ: " + physicalType);
-        //        }
-        //        break;
+      case TIMESTAMP_TZ:
+        switch (physicalType) {
+          case SB8:
+            {
+              arrowType = Types.MinorType.STRUCT.getType();
+              FieldType fieldTypeEpoch =
+                  new FieldType(true, Types.MinorType.BIGINT.getType(), null, metadata);
+              FieldType fieldTypeTimezone =
+                  new FieldType(true, Types.MinorType.INT.getType(), null, metadata);
+              Field fieldEpoch = new Field(FIELD_EPOCH_IN_SECONDS, fieldTypeEpoch, null);
+              Field fieldTimezone = new Field(FIELD_TIME_ZONE, fieldTypeTimezone, null);
+
+              children = new LinkedList<>();
```

Comment:
Does this have to be a linked list? Can you do something like `children = Arrays.of(fieldEpoch, fieldTimezone)`?

Reply:
The Field [constructor](https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java#L102) expects a List, I can make it an ArrayList though.